sympy/stats/stochastic_process_types.py

```diff
@@ -2181,8 +2196,11 @@ def lamda(self):
     def state_space(self):
         return S.Naturals0
 
-    def distribution(self, rv):
-        return PoissonDistribution(self.lamda*rv.key)
+    def distribution(self, key=None):
```

Comment:
I am not sure if `key=None` should be kept here or not due to function signatures.

Reply:
Either seems fine to me, so I will go with your suggestion on this.

Reply:
If we  allow `key=None`, then it may lead to errors due to absence of `key` attribute in `None`. I think a simple, `key` without defaulting to `None` should also work. Can you check?